randovania/gui/lib/qt_network_client.py

```diff
@@ -117,51 +169,48 @@ def available_login_methods(self) -> set[str]:
             methods.append("discord")
         return set(methods)
 
+    @asyncSlot()
+    @handle_network_errors
+    async def attempt_join(self, session: MultiplayerSessionListEntry):
+        if session.has_password and not session.is_user_in_session:
+            dialog = QInputDialog(self)
+            dialog.setWindowTitle("Enter password")
+            dialog.setLabelText("This session requires a password:")
+            dialog.setWindowModality(Qt.WindowModal)
+            dialog.setTextEchoMode(QLineEdit.EchoMode.Password)
 
-def handle_network_errors(fn):
-    @functools.wraps(fn)
-    async def wrapper(self, *args, **kwargs):
-        try:
-            return await fn(self, *args, **kwargs)
-
-        except InvalidAction as e:
-            await async_dialog.warning(self, "Invalid action", f"{e}")
-
-        except ServerError:
-            await async_dialog.warning(self, "Server error",
-                                       "An error occurred on the server while processing your request.")
+            if await async_dialog.execute_dialog(dialog) != QDialog.DialogCode.Accepted:
+                return
 
-        except NotLoggedIn:
-            await async_dialog.warning(self, "Unauthenticated",
-                                       "You must be logged in.")
+            password = dialog.textValue()
+        else:
+            password = None
 
-        except NotAuthorizedForAction:
-            await async_dialog.warning(self, "Unauthorized",
-                                       "You're not authorized to perform that action.")
+        try:
+            joined_session = await self.join_multiplayer_session(session, password)
+            return joined_session
 
-        except UserNotAuthorized:
-            await async_dialog.warning(
-                self, "Unauthorized",
-                "You're not authorized to use this build.\nPlease check #dev-builds for more details.",
-            )
+        except WrongPassword:
+            await async_dialog.warning(self, "Incorrect Password", "The password entered was incorrect.")
 
-        except UnsupportedClient as e:
-            s = e.detail.replace('\n', '<br />')
-            await async_dialog.warning(
-                self, "Unsupported client",
-                s,
-            )
+    async def ensure_logged_in(self, parent: QWidget | None):
```

Comment:
I don't really like this being here.
In particular I'm planning to have "weak" and "strong" auth. 

When you restore a session, if your discord re-auth fails you'll be "weak" auth. You can still do world sync and play multi, but can't open and session gui.
Explicitly doing a login flow always grants "strong" auth.

Reply:
If we remove the call to that function in the section of the first comment, it get's only called by `_host_game_session`, `_browse_for_session` and `_attempt_join` (pressing the new action in the game connection window).
That sounds fine to me because that is where you want the strong auth.

Reply:
I'd just remove the CLI command for joining a session directly from `qt.py`

Reply:
That works but dev builds need to call the login method. I only broke every future dev build that's why I thought I maybe should fix it :p